### ISSUE TITLE
MySQL parser fails on double parentheses around nested join

### ISSUE THREAD

**Issue by gunnaruw** [sqlfluff/sqlfluff#8382]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Running `sqlfluff lint --dialect mysql` on this MySQL query (simplified for reproduction)

```mysql
SELECT 1 FROM a LEFT JOIN ((b INNER JOIN c ON TRUE)) ON TRUE;
```

yields a parse error

```sh
sqlfluff lint --dialect mysql sql/views/test.sql
== [sql/views/test.sql] FAIL                                                                                                                                                                 
L:   1 | P:  17 |  PRS | Line 1, Position 17: Found unparsable section: 'LEFT
                       | JOIN ((b INNER JOIN c ON TRUE)) ON ...'
WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

But this query is valid MySQL and executes successfully on my MySQL 8.4.10 server when replacing a, b, and c with real table names.

If you remove the extra parentheses around the right-hand side of the join:

```mysql
SELECT 1 FROM a LEFT JOIN (b INNER JOIN c ON TRUE) ON TRUE;
```

the parsing error doesn't occur and it reports correctly that the query fails the unused_join rule:

```sh
sqlfluff lint --dialect mysql sql/views/test.sql
== [sql/views/test.sql] FAIL                                                                                                                                                                 
L:   1 | P:  15 | ST11 | Joined table 'a' not referenced elsewhere in query
                       | [structure.unused_join]
L:   1 | P:  27 | ST11 | Joined table '(b INNER JOIN c ON TRUE)' not referenced
                       | elsewhere in query [structure.unused_join]
All Finished 📜 🎉!
```

### Expected Behaviour

The mysql dialect parser should not fail to parse due to a set of redundant parentheses around a nested join.

### Observed Behaviour

Parsing error on valid MySQL.

### How to reproduce

```bash
printf 'SELECT 1 FROM a LEFT JOIN ((b INNER JOIN c ON TRUE)) ON TRUE;\n' | sqlfluff lint - --dialect mysql
```

will yield

```bash
== [stdin] FAIL                                                                                                                                                                              
L:   1 | P:  17 |  PRS | Line 1, Position 17: Found unparsable section: 'LEFT
                       | JOIN ((b INNER JOIN c ON TRUE)) ON ...'
WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Dialect

mysql

### Version

sqlfluff, version 4.3.0

Python 3.11.2; also tested on Python 3.14.7.

### Configuration

No .sqlfluff file

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by devYRPauli** [comment]
This is not a mysql defect. The grammar that fails is `FromExpressionElementSegment` in `src/sqlfluff/dialects/dialect_ansi.py`, and mysql does not override it.

```python
match_grammar: Matchable = OneOf(
    _base_from_expression_element,
    Bracketed(
        Sequence(
            _base_from_expression_element,
            AnyNumberOf(Ref("JoinClauseSegment")),
        ),
    ),
)
```

The bracketed branch holds `_base_from_expression_element`, not a reference back to itself. So one pair of brackets around a nested join matches and two pairs do not.

I ran your two queries plus a three-bracket version on main at d3e8358:

| dialect | `(b JOIN c)` | `((b JOIN c))` | `(((b JOIN c)))` |
|---|---|---|---|
| ansi | parses | PRS | PRS |
| mysql | parses | PRS | PRS |
| mariadb | parses | PRS | PRS |
| postgres | parses | PRS | PRS |
| sqlite | parses | PRS | PRS |
| bigquery | parses | PRS | PRS |
| duckdb | parses | PRS | PRS |
| tsql | parses | parses | parses |

tsql passes because it overrides `TableExpressionSegment` with a bracketed branch that refers to `TableExpressionSegment` again. Its docstring names the reason: "table expressions ... can also be join tables, optionally bracketed, allowing for nested joins".

The same shape in ansi accepts any depth. I have that change on a branch and I am running the dialect fixture suite against it now. I will open a PR if it stays green.

**Comment by devYRPauli** [comment]
PR is #8457.

It does not use the tsql shape I described above. I tried that first: a `TableExpressionSegment` override with a `Bracketed(Sequence(Ref("TableExpressionSegment"), ..., Ref("JoinClauseSegment")))` branch. It parses every depth, but it also rewrites the tree for SQL that already parsed. On `test/fixtures/dialects/ansi/select_nested_join.sql` the parse tree changed by 370 lines, `sqlfluff lint` gained 8 LT02 violations, dropped 2 ST09 findings and raised 1 elsewhere, and `sqlfluff fix` wrote different output. I dropped that shape.

The change I opened points the join target at `FromExpressionSegment` for the bracketed case only:

```python
_join_target = OneOf(
    Ref("FromExpressionElementSegment"),
    Bracketed(Ref("FromExpressionSegment")),
)
```

`FromExpressionSegment` already recurses through its own bracketed branch, added in #4890, so the extra pairs need no new recursion. Both existing fixtures diff to 0 lines with this shape.

One consumer needed a fix as well. `get_join_clause_aliases` in `src/sqlfluff/core/dialects/common.py` reads the join clause's direct `from_expression_element` child and asserts it exists. Under the new branch that element sits inside the brackets, so eight rules raised `AssertionError` on valid SQL: AL04, AL05, AM04, RF01, RF02, RF03, ST05 and ST09. The PR adds a bounded `recursive_crawl` there.

**Comment by devYRPauli** [comment]
Correction to my two comments above: #8393 already fixes this and has been open since 24 August. I missed it and opened a duplicate, #8457, which I have closed. #8393 is under review with @peterbud and covers more dialects than mine did.
